pydis_site/apps/api/viewsets/bot/mailing_list.py

```diff
@@ -0,0 +1,85 @@
+from django.db import IntegrityError
+from rest_framework import status
+from rest_framework.decorators import action
+from rest_framework.exceptions import ParseError
+from rest_framework.mixins import ListModelMixin, RetrieveModelMixin
+from rest_framework.request import Request
+from rest_framework.response import Response
+from rest_framework.viewsets import GenericViewSet
+
+from pydis_site.apps.api.serializers import MailingListSerializer
+from pydis_site.apps.api.models import MailingList, MailingListSeenItem
+
+
+class MailingListViewSet(GenericViewSet, ListModelMixin, RetrieveModelMixin):
+    """
+    View providing management and updates of mailing lists and their seen items.
+
+    ## Routes
+
+    ### GET /bot/mailing-lists
+    Returns all the mailing lists and their seen items.
+
+    #### Response format
+    >>> [
+    ...     {
+    ...         'id': 1,
+    ...         'name': 'python-dev',
+    ...         'seen_items': [
```

Comment:
Is there a reason why you're returning a list of objects instead of a list of the hashes directly ?

Reply:
Yes. I was lazy :smiley:

I've changed and pushed this.